specification/resources/resource-manager/Microsoft.Resources/preview/2023-03-01-preview/changes.json

```diff
@@ -0,0 +1,424 @@
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/ChangeProperties"
+        }
+      }
+    },
+    "ChangeProperties": {
+      "description": "The properties of a change",
+      "type": "object",
+      "properties": {
+        "targetResourceId": {
+          "description": "The fully qualified ID of the target resource that was changed",
+          "type": "string",
+          "readOnly": true
+        },
+        "targetResourceType": {
+          "description": "The namespace and type of the resource",
+          "type": "string",
+          "readOnly": true
+        },
+        "changeType": {
+          "description": "The type of change that was captured in the resource",
+          "type": "string",
+          "readOnly": true,
+          "enum": [
+            "Update",
+            "Delete",
+            "Create"
+          ],
+          "x-ms-enum": {
+            "name": "ChangeType",
+            "modelAsString": false,
+            "values": [
+              {
+                "value": "Update",
+                "description": "An existing resource underwent a change",
+                "name": "Update"
+              },
+              {
+                "value": "Delete",
+                "description": "An existing resource was deleted",
+                "name": "Delete"
+              },
+              {
+                "value": "Create",
+                "description": "A newly created resource",
+                "name": "Create"
+              }
+            ]
+          }
+        },
+        "changeAttributes": {
+          "$ref": "#/definitions/ChangeAttributes"
+        },
+        "changes": {
+          "$ref": "#/definitions/ChangesDictionary"
+        }
+      }
+    },
+    "ChangeAttributes": {
+      "description": "Details about the change resource",
+      "type": "object",
+      "properties": {
+        "correlationId": {
+          "description": "The ARM correlation ID of the change resource",
+          "type": "string",
+          "readOnly": true
+        },
+        "timestamp": {
+          "description": "The time the change(s) on the target resource ocurred",
+          "type": "string",
+          "readOnly": true
+        },
+        "changesCount": {
+          "type": "integer",
+          "format": "int64",
+          "readOnly": true,
+          "description": "The number of changes this resource captures"
+        },
+        "previousResourceSnapshotId": {
+          "description": "The GUID of the previous snapshot",
+          "type": "string",
+          "readOnly": true
+        },
+        "newResourceSnapshotId": {
+          "description": "The GUID of the new snapshot",
+          "type": "string",
+          "readOnly": true
+        },
+        "isTruncated": {
+          "description": "The flag indicating whether property changes dictionary was truncated in this resource",
+          "type": "boolean",
+          "readOnly": true
```

Comment:
consider using an enum instead : 

[https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices](https://eng.ms/docs/products/arm/rp_onboarding/process/api_review_best_practices)



Replace boolean/switch properties with better enum
A Boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Even if you still believe [True, False] are the correct values for a property, you should use a string enum with values [True, False] instead of boolean. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [True, False, Unknown].